rad/prelude/io.rad

```diff
@@ -260,6 +260,16 @@ process exited normally and `[:error n]` otherwise. Example: `(process!
     (write-handle! (stderr!) prompt)
     (read-line-handle! (stdin!))))
 
+(def-rec prompt-non-empty!
```

Comment:
This doesn't seem to be a very general function. Ideally the abstraction is something that takes a validator, an error message, and repeats the prompt until the validator passes.